src/thor/triplegbuilder.cc

```diff
@@ -1042,6 +946,328 @@ void AccumulateRecostingInfoForward(const valhalla::Options& options,
   }
 }
 
+/**
+ * A simple way to encapsulate the marshalling of native data types specific to multi modal into
+ * the protobuf ones that we pass downstream to odin and other clients
+ */
+struct MultimodalBuilder {
```

Comment:
Optional: my only suggestion would be to move this struct to a new file, it would be nice to trim down the number of lines in this file while you're doing some cleanup. Not a blocker though.